projects/core/src/lib/layer-collections/layer-collection.service.ts

```diff
@@ -45,4 +76,186 @@ export class LayerCollectionService {
             .getSessionStream()
             .pipe(mergeMap((session) => this.backend.getWorkflowMetadata(workflowId, session.sessionToken)));
     }
+
+    /**
+     * Add all layers (directly) contained in a layer collection to the current project.
+     */
+    addCollectionLayersToProject(collection: LayerCollectionDict): void {
+        const layersObservable = collection.items
+            .filter((layer) => layer.type === 'layer')
+            .map((layer) => layer as LayerCollectionLayerDict)
+            .map((layer) => this.resolveLayer(layer.id));
+
+        // TODO: lookup in parallel
```

Comment:
mergeMap?

Reply:
my idea was to add the layers as soon as the metadata is retrieved instead of waiting for all to be finished, but I don't know if that really makes any difference.

Reply:
Currently, there is the danger to have a race condition when calling addLayer concurrently I guess. At least we would have to look into this.

With mergeMap, we would have at least all called concurrently, while for zip they are called in order or am I mistaking?

Reply:
Where do you see a race condition? I thought JavaScript only uses a single thread anyway.

> With mergeMap, we would have at least all called concurrently, while for zip they are called in order or am I mistaking?

no, zip only emits after all observables have emitted. But: we could start adding the first layer as soon as it (or rather: its metdata) is ready. I wanted to use `forkJoin` for this but couldn't get it to work, so I left a TODO here.